### ISSUE TITLE
`document.monetization` vs global

### ISSUE THREAD

**Issue by adrianhopebailie** [WICG/webmonetization#24]
From web-monetization created by [marcoscaceres](https://github.com/marcoscaceres): adrianhopebailie/web-monetization#23  

Wondering what the rationale was for attaching monetization to the `Document` object? It doesn't really feel like the right place to attach this, as monetization doesn't itself affect the document or have anything specifically to do with the document interface. We should consider maybe having monetization either on Navigator or its own `Monetization` namespace.

**Comment by adrianhopebailie** [comment]
The decision about where to put this was not strongly sided either way. If it's better as a stand-alone global then let's make it a global.

**Comment by adrianhopebailie** [comment]
@marcoscaceres 
Curious as to why that is.
Is that just the prevailing convention, or you have a rationale? or both?? :)

**Comment by adrianhopebailie** [comment]
Yes, putting something in the global namespace would be my preference.

**Comment by adrianhopebailie** [comment]
And here:
https://github.com/interledger/rfcs/pull/500#issuecomment-452932589
And:
https://github.com/interledger/rfcs/pull/500#issuecomment-453173372

To quote @justmoon from comments above:

> To give some context: This made more sense to me because Web Monetization is now triggered by a meta-tag and meta-tags describe the document. So whereas previously it was a feature of the JavaScript environment (hence existing on window, it's now a feature of the HTML document.

**Comment by adrianhopebailie** [comment]
Yeah, I'm not sure. I wasn't really around for the beginnings of the spec, but I do recall reading some discussion when starting up. Tracked it down to here: https://github.com/interledger/rfcs/pull/500#discussion_r245246937

**Comment by adrianhopebailie** [comment]
Thanks again for the info :)

**Comment by adrianhopebailie** [comment]
> Is that just the prevailing convention, or you have a rationale? or both?? :)

Bit of both... we generally like to put things on the document that directly affect the Document.  You can see what I mean by looking at the interface definition:

https://html.spec.whatwg.org/#the-document-object

In most cases, the methods and attributes affect the Document **directly** in some meaningful way. Monetization doesn't affect the Document or reflect some sate of the document, so should probably not be hung on that interface.

**Comment by adrianhopebailie** [comment]
## PROPOSAL

Based on the discussion so far we should move the `Monetization` interface off of `document` and make it a global.

We can leave the existing `document.monetization` interface for backwards compatibility but mark it as deprecated.

This would also allow us to define a new API shape that takes into account the various other suggestions such as:

- changing the event names to conform to standard practice
- merging the event types and not use `CustomEvent` ( #23 )
- adding a `ReadableStream` that can be used to consume payments ( #27 and #25 )
- using the existing `PaymentAmountCurrency` interface ( #16 )

Please respond to this proposal with either an 👍 or a 👎 

If you are 👎 on this proposal please provide rationale below

**Comment by sublimator** [comment]
End of the day tired, but I'm in favor of this:

I can't imagine doing the donkey work of getting this working being *that* hard.
We just need to decide to move in this direction.

For:

1. Making whatever we see as enduring changes as soon as possible would mean less
changes later.
2. We wouldn't have to change the meaning of `pending` on
document.monetization, which is a subtly breaking change.

Against:

1. I can't really see any downside, except that it's not exactly an ongoing release/versioning strategy so much as a one time opportunity.

Generally, the end users (i.e. extension users) are going to have the most up-to-date extension released due to automatic updates.

Integrators are going to have a less easy time of it. Putting some thought into release management/explicit api versioning might help us (document.monetization ~= v2, window.monetization ~= v3 is only implied)

e.g. 
```javascript
// the first thing that comes to mind, but an illustration of what I mean
window.monetization.draft1.monetizationStream(...) 
```

**Comment by sharafian** [comment]
I have a preference for holding off these renaming-type changes for the last minute so that we can group them all together

**Comment by adrianhopebailie** [comment]
CONCLUSION: Move `monetization` to global

**Comment by sublimator** [comment]
https://github.com/sidvishnoi/gecko-webmonetization/commit/517b8ce6b1f7fa3bd1679dbc929145d80390f387#commitcomment-46412865

**Comment by marcoscaceres** [comment]
it still feels safer to put it on `Navigator`, tbh (to avoid any potential collisions with existing user-land code). I think there is also W3C TAG guidance around this... though I can't remember where.   

Sorry also about not responding above. There were other issues too, with `Document`... like a page can have/create multiple `Documents` (e.g., via `document.implementation.createDocument()`). 

Anyway, the point in kinda mute if we are willing to shift it elsewhere.

**Comment by mankins** [comment]
`Navigator` makes sense to me.

I do wonder if `monetization` is actually too broad and it should be `webmon` or something more specific instead? There are multiple different ways to monetize a site (and multiple ways to spell monetise) ...

**Comment by huijing** [comment]
Resolved by #193
